src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java

```diff
@@ -2692,4 +2694,323 @@ public void terminate() throws Exception {
             }
         }
     }
+    
+    
+    @Nested
+    public class TCGenKeys {
+        private final String tableName = AbstractSQLGenerator
+                .escapeIdentifier(RandomUtil.getIdentifier("TCInsertWithGenKeys"));
+        private final String idTableName = AbstractSQLGenerator
+                .escapeIdentifier(RandomUtil.getIdentifier("TCInsertWithGenKeysIDs"));
+
+        private final String triggerName = AbstractSQLGenerator.escapeIdentifier("Trigger");
+        private final int NUM_ROWS = 3;
+
+        @BeforeEach
+        public void setup() throws Exception {
+            try (Connection con = getConnection()) {
+                con.setAutoCommit(false);
+                try (Statement stmt = con.createStatement()) {
+                    TestUtils.dropTriggerIfExists(triggerName, stmt);
+                    stmt.executeUpdate("CREATE TABLE " + tableName + " (ID int NOT NULL IDENTITY(1,1) PRIMARY KEY, NAME varchar(32));");
+
+                    stmt.executeUpdate("CREATE TABLE " + idTableName  + "(ID int NOT NULL IDENTITY(1,1) PRIMARY KEY);");
+
+                    stmt.executeUpdate("CREATE TRIGGER " + triggerName + " ON " + tableName + " FOR INSERT AS INSERT INTO " + idTableName + " DEFAULT VALUES;");
+
+                    for (int i = 0; i < NUM_ROWS; i++) {
+                        stmt.executeUpdate("INSERT INTO " + tableName + " (NAME) VALUES ('test')");
+                    }
+
+                }
+                con.commit();
+            }
+        }
+
+        /**
+         * Tests executeUpdate for Insert followed by getGenerateKeys
+         *
+         * @throws Exception
+         */
+        @Test
+        public void testExecuteUpdateInsertAndGenKeys() throws Exception {
```

Comment:
Check if these throws can be removed. For many of these methods no Exception is actually thrown.
```suggestion
        public void testExecuteUpdateInsertAndGenKeys() {
```

Reply:
done